manipulation.go

```diff
@@ -572,3 +652,24 @@ func (s *Selection) manipulateNodes(ns []*html.Node, reverse bool,
 
 	return s
 }
+
+func (s *Selection) eachNodeHtml(htmlStr string, parent bool, fn func(n *html.Node, nodes []*html.Node)) *Selection {
```

Comment:
This would require a bit of documentation, explaining the purpose (in general, but also specifically of the nodesMap).